cluster/manifests/01-routegroup/routegroup-crd.yaml

```diff
@@ -145,9 +145,9 @@ spec:
                 description: List of hostnames for the RouteGroup
                 items:
                   pattern: "^[a-z0-9]([-a-z0-9]*[a-z0-9])?([.][a-z0-9]([-a-z0-9]*[a-z0-9])?)*$"
-                  minItems: 1 # original CRD allows empty hosts but we explicitly want it specified in our infrastructure
                   type: string
                 type: array
+                minItems: 1
```

Comment:
Consider
```yaml
apiVersion: zalando.org/v1
kind: RouteGroup
metadata:
  name: no-hosts
spec:
  backends:
    - name: example
      type: network
      address: https://example.org
  defaultBackends:
    - backendName: example
---
apiVersion: zalando.org/v1
kind: RouteGroup
metadata:
  name: empty-hosts
spec:
  backends:
    - name: example
      type: network
      address: https://example.org
  defaultBackends:
    - backendName: example
  hosts: []

```

when `.Cluster.ConfigItems.skipper_ingress_routegroup_crd_require_hosts` is "true" (default):
```
$ kubectl apply -f rg.yaml 
Error from server (Invalid): error when creating "rg.yaml": RouteGroup.zalando.org "no-hosts" is invalid: spec.hosts: Required value
Error from server (Invalid): error when creating "rg.yaml": RouteGroup.zalando.org "empty-hosts" is invalid: spec.hosts: Invalid value: 0: spec.hosts in body should have at least 1 items
```

when `.Cluster.ConfigItems.skipper_ingress_routegroup_crd_require_hosts` is "false":
```
$ kubectl apply -f rg.yaml 
routegroup.zalando.org/no-hosts created
The RouteGroup "empty-hosts" is invalid: spec.hosts: Invalid value: 0: spec.hosts in body should have at least 1 items
```